marlowe-playground-client/src/HaskellEditor.purs

```diff
@@ -43,18 +45,31 @@ render state =
               (demoScriptLink <$> Array.fromFoldable (Map.keys StaticData.demoFiles))
           ]
       ]
-  , section [ classes (haskellEditor state) ]
-      [ mapComponent
-          HaskellEditorAction
-          $ editorView defaultContents _haskellEditorSlot StaticData.bufferLocalStorageKey editorPreferences
+  , section [ class_ (ClassName "code-panel") ]
+      [ div [ classes (codeEditor state) ]
+          [ haskellEditor state ]
       ]
   ]
   where
   demoScriptLink key = li [ classes (isActiveDemo state) ] [ a [ onClick $ const $ Just $ LoadHaskellScript key ] [ text key ] ]
 
-  editorPreferences = view _editorPreferences state
+haskellEditor ::
+  forall m.
+  MonadAff m =>
+  FrontendState ->
+  ComponentHTML HAction ChildSlots m
+haskellEditor state = slot _haskellEditorSlot unit component unit (Just <<< HaskellHandleEditorMessage)
+  where
+  setup editor = do
+    mContents <- liftEffect $ LocalStorage.getItem StaticData.bufferLocalStorageKey
+    let
+      contents = fromMaybe initialContents mContents
+    model <- liftEffect $ Monaco.getModel editor
+    liftEffect $ Monaco.setValue model contents
 
-  defaultContents = Map.lookup "Escrow" StaticData.demoFiles
+  component = monacoComponent $ HM.settings setup
+
+  initialContents = fromMaybe "" $ Map.lookup "Escrow" StaticData.demoFiles
```

Comment:
How about:

``` purescript
  let contents = fromMaybe "" $ initialContents <|> mContents
...
initialContents = Map.lookup "Escrow" StaticData.demoFiles
```

Throws away a bit of noise and puts the default where it's most usefully seen, I think.

Reply:
I realised I can just import the original escrow string rather than get it from the map so I have `fromMaybe HE.escrow mContents`